### ISSUE TITLE
false-positive: no-value-for-parameter

### ISSUE THREAD

**Issue by ikkoham** [PyCQA/pylint#8325]
### Bug description

```python
class Test:
    """Test"""

    def test(self):
        """Test"""
        self.__new__(type(self))  # NG
        type(self).__new__(type(self))  # OK
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:9:8: E1120: No value for argument 'cls' in classmethod call (no-value-for-parameter)
```

### Expected behavior

Both should be ok.

### Pylint version

```shell
pylint 2.16.2
```

**Comment by AriHC** [comment]
I've noticed this occurring on calls to `__new__` as well
